langchain/vectorstores/pinecone.py

```diff
@@ -68,14 +73,57 @@ def add_texts(
             docs.append((id, embedding, metadata))
             ids.append(id)
         # upsert to Pinecone
-        self._index.upsert(vectors=docs)
+        self._index.upsert(vectors=docs, namespace=namespace)
         return ids
 
-    def similarity_search(self, query: str, k: int = 5) -> List[Document]:
-        """Look up similar documents in pinecone."""
+    def similarity_search_with_score(
```

Comment:
i'd also imagine this is a pretty common use case, could be worth adding it as an abstractmethod in the base VectorStore class

Reply:
yup youre absolutely right! I added to FAISS just to start because that was the immediate use case, but absolutely right that it should be